ai_server/server.py

```diff
@@ -1,12 +1,250 @@
 from flask import Flask, request, jsonify, abort
 import ollama
+import subprocess
+import os
+import requests
+from typing import Optional
+from dotenv import load_dotenv
 
 from .redis_helper import REDIS_CONNECTION
 
+# Load environment variables from .env file
+load_dotenv()
+
 app = Flask('AI server')
 
 DEFAULT_MODEL = 'deepseek-coder-v2:latest'
 
+# Llama.cpp configuration
+LLAMA_CPP_CLI = '/data1/llama.cpp/bin/llama-cli'
+GGUF_DIR = '/data1/GGUF'
+
+# Llama server configuration - now uses environment variable
+LLAMA_SERVER_URL = os.getenv('LLAMA_SERVER_URL')  # e.g., http://localhost:8080
+
+# Known llama.cpp model in /data1/GGUF/
+# We could add more models here if needed.
+LLAMA_CPP_MODEL_DIRS = [
+    'DeepSeek-V2.5-IQ1_M',
+    'DeepSeek-V3-0324-UD-IQ2_XXS', 
+    'DeepSeek-V3-0324-UD-Q2_K_XL'
+]
+
+# Output parsing skip patterns for llama-cli
+LLAMA_CPP_SKIP_PATTERNS = [
```

Comment:
I'm not convinced this is necessary; `llama-cli` does print out a bunch of debugging information, but I believe it all goes to stderr, not stdout.